"first commit"

psasha007 opened the pull request.

psasha007 committed:
"first commit"

psasha007 committed:
Добавил класс BasePage

psasha007 committed:
Добавил класс BaseTest

psasha007 committed:
Добавил класс Constants

psasha007 committed:
Добавил класс EntrancePageObject - PageObject страницы авторизации

psasha007 committed:
Добавил класс EntranceTest, с проверками
вход по кнопке «Войти в аккаунт» на главной,
вход через кнопку «Личный кабинет»,
вход через кнопку в форме регистрации,
вход через кнопку в форме восстановления пароля.

psasha007 committed:
Добавил класс RegistrationPageObject - PageObject страницы регистрации

psasha007 committed:
Добавил классы RegistrationIncorrectPasswordParameterizedTest,    RegistrationTrueTest с проверками
Успешной регистраций
и на Ошибку для некорректного пароля. Минимальный пароль — шесть символов.

psasha007 committed:
Добавил класс SectionConstructorPageObject - PageObject раздела «Конструктор»

psasha007 committed:
Добавил класс SectionConstructorTest с проверками
переходов к разделам:
«Булки»,
«Соусы»,
«Начинки».

psasha007 committed:
Добавил класс PersonalAccountPageObject - PageObject основная страница для авторизации и т.д.

psasha007 committed:
Добавил класс GoToPersonalAccountTest с поверкой
перехода по клику на «Личный кабинет»

psasha007 committed:
Добавил класс GoToPersonalAccountTest с поверкой
выхода по кнопке «Выйти» в личном кабинете

psasha007 committed:
Добавил класс TransitionToDesignerTest с поверками
переход по клику на «Конструктор» и на логотип Stellar Burgers

psasha007 committed:
Добавил allure-results

nikolay-977 commented on src/test/java/pageobject/BasePage.java:
```diff
@@ -0,0 +1,30 @@
+package pageobject;
```
Необходимо исправить: Перенеси page objects в src.main.java.pageobjects

psasha007 replied:
Перенес page objects в src.main.java.pageobjects

nikolay-977 commented on src/test/java/pageobject/PersonalAccountPageObject.java:
```diff
@@ -0,0 +1,161 @@
+    }
+
+    // Конструктор
+    public PersonalAccountPageObject(WebDriver driver) {
```
Необходимо исправить: В начале класса следует описывать переменные, затем конструкторы, а после конструкторов - методы. Не следует в Page Objects делать иначе. Так код будет легче читаемым. Это упростит его поддержку.

psasha007 replied:
понял, учел и поправил

nikolay-977 commented on src/test/java/pageobject/RegistrationPageObject.java:
```diff
@@ -0,0 +1,88 @@
+    public RegistrationPageObject checkTextIncorrectPassword(String expected){
+        waitForElement(textIncorrectPassword, Constants.TIME_OUT_IN_SECONDS);
+        WebElement actual = driver.findElement(textIncorrectPassword);
+        Assert.assertEquals("Ошибка, текст не совпадает: ", expected, actual.getText());
```
Можно улучшить: в информационном сообщении лучше указывать суть проверки. Это сократит время локализации дефекта.

psasha007 replied:
улучшил сообщения

psasha007 committed:
- Осуществил перенос page objects в src.main.java.pageobjects
- Перенес конструкторы в соответствии требованиями:
В начале класса описывать переменные
затем конструкторы, а после конструкторов - методы
- Улучшил в информационные сообщения, теперь указываю суть проверки
